internal/controllers/addoninstance/addon_instance_controller.go

```diff
@@ -0,0 +1,127 @@
+package addoninstance
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	av1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	addoninstance "github.com/openshift/addon-operator/pkg/client"
+	rv1alpha1 "github.com/openshift/reference-addon/apis/reference/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// Default timeout when we do a manual RequeueAfter
+	defaultRetryAfterTime = 10 * time.Second
```

Comment:
We can move this into a config option to facilitate testing and invert control to the initializer of this controller.